.github/workflows/mingw.yml

```diff
@@ -0,0 +1,229 @@
+          srcdir=.
+
+          # now install things
+
+          pkgdir=python_pkgdir
+
+          PREFIX_WIN=$(cygpath -wm ${MINGW_PREFIX})
+          MSYSTEM=MINGW \
+          MSYS2_ARG_CONV_EXCL="--prefix=;--install-scripts=;--install-platlib=" \
+          make -j1 install DESTDIR="${pkgdir}"
+          VERABI=${_pybasever}
+
+          # gdb pretty printers for debugging Python itself; to use:
+          # python
+          # sys.path.append('C:/msys64/mingw64/share/gdb/python3')
+          # import python_gdb
+          # reload(python_gdb)
+          # end
+
+          cp -f "${pkgdir}${MINGW_PREFIX}"/lib/python${_pybasever}/config-${VERABI}/libpython${VERABI}.dll.a "${pkgdir}${MINGW_PREFIX}"/lib/libpython${VERABI}.dll.a
+
+          # Need for building boost python module
+          cp -f "${pkgdir}${MINGW_PREFIX}"/lib/python${_pybasever}/config-${VERABI}/libpython${VERABI}.dll.a "${pkgdir}${MINGW_PREFIX}"/lib/libpython${_pybasever}.dll.a
+
+          for fscripts in 2to3 2to3-${_pybasever} idle3 idle${_pybasever} pydoc3 pydoc${_pybasever}; do
+              sed -i "s|$(cygpath -w ${MINGW_PREFIX} | sed 's|\\|\\\\|g')/bin/python${_pybasever}.exe|/usr/bin/env python${_pybasever}.exe|g" "${pkgdir}${MINGW_PREFIX}"/bin/${fscripts}
+          done
+
+
+          cp "${pkgdir}${MINGW_PREFIX}"/bin/python3.exe "${pkgdir}${MINGW_PREFIX}"/bin/python.exe
+          cp "${pkgdir}${MINGW_PREFIX}"/bin/python3w.exe "${pkgdir}${MINGW_PREFIX}"/bin/pythonw.exe
+          cp "${pkgdir}${MINGW_PREFIX}"/bin/python3-config "${pkgdir}${MINGW_PREFIX}"/bin/python-config
+          cp "${pkgdir}${MINGW_PREFIX}"/bin/idle3 "${pkgdir}${MINGW_PREFIX}"/bin/idle
+          cp "${pkgdir}${MINGW_PREFIX}"/bin/pydoc3 "${pkgdir}${MINGW_PREFIX}"/bin/pydoc
+
+          sed -i "s|#!${pkgdir}${MINGW_PREFIX}/bin/python${VERABI}.exe|#!/usr/bin/env python${_pybasever}.exe|" "${pkgdir}${MINGW_PREFIX}"/lib/python${_pybasever}/config-${VERABI}/python-config.py
+
+          # fix permissons
+          find ${pkgdir}${MINGW_PREFIX} -type f \( -name "*.dll" -o -name "*.exe" \) | xargs chmod 0755
+
+          # replace paths in sysconfig
+          sed -i "s|${pkgdir}${MINGW_PREFIX}|${MINGW_PREFIX}|g" \
+          "${pkgdir}${MINGW_PREFIX}"/lib/python${_pybasever}/lib-dynload/_sysconfigdata*.py \
+          "${pkgdir}${MINGW_PREFIX}"/lib/python${_pybasever}/smtpd.py
+
+          # install venv launchers
+          mkdir -p "${pkgdir}${MINGW_PREFIX}"/lib/python${_pybasever}/venv/scripts/nt
+          cp venvlauncher.exe "${pkgdir}${MINGW_PREFIX}"/lib/python${_pybasever}/venv/scripts/nt/python.exe
+          cp venvwlauncher.exe "${pkgdir}${MINGW_PREFIX}"/lib/python${_pybasever}/venv/scripts/nt/pythonw.exe
+
+      - name: Run Smoke Test
+        shell: msys2 {0}
+        run: |
+          cd python_pkgdir/${MINGW_PREFIX}/bin
+          curl -OL https://gist.github.com/naveen521kk/cbbdc838a12a407711a454bb1bfe8b75/raw/a0702cabb8805dade2a0c23234e8e8dc0e1ef8d0/smoketests.py
```

Comment:
is it fine to get them from the gist or should I add that file here?

Reply:
if I should add where should it be?

Reply:
hm, maybe add under to ".github/workflows"?